### PR TITLE
Require Jenkins 2.479 or newer

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Require Jenkins 2.479 or newer

Update to use Jakarta EE 9, StaplerRequest2, and StaplerResponse2

### Testing done

Automated tests pass.  Once the CI job has created an incremental build, I'll inclide it in my test environment for interactive checks.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/plugins/git/GitStatus.java [hunk 1]
```diff
@@ -115,6 +115,75 @@ public String toString() {
         return s.toString();
     }
 
+    /**
+     * Copied from scm API plugin.
```
Comment (basil) [inline src/main/java/hudson/plugins/git/GitStatus.java:119]: This is not maintainable; better to depend on https://github.com/jenkinsci/scm-api-plugin/pull/282 which migrates this method (compatibly!) to EE 9.

**Comment by basil** [comment]
Worth a BOM/PCT run I think to ensure that no other plugins are consuming the changed methods. From a visual inspection they looked like one-off forms that are unlikely to be reused, but one never knows.

**Review by olamy — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -114,6 +118,8 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>scm-api</artifactId>
+      <!-- TODO: Replace with release once it is available -->
+      <version>698.v29a_7e2c2a_6cf</version>
```
Comment (basil) [inline pom.xml:122]:
```suggestion
      <!-- TODO: Replace with release of https://github.com/jenkinsci/scm-api-plugin/pull/282 once it is available -->
      <version>698.v1b_46fc0633b_9</version>
```

**Review by basil — APPROVED** [review]
Thanks!

File: pom.xml [hunk 3]
```diff
@@ -114,6 +118,8 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>scm-api</artifactId>
+      <!-- TODO: Replace with release of https://github.com/jenkinsci/scm-api-plugin/pull/282 once it is available -->
```
Comment (jglick) [inline pom.xml:121]: https://github.com/jenkinsci/scm-api-plugin/releases/tag/698.v8e3b_c788f0a_6?

Comment (MarkEWaite) [reply pom.xml]: 87fcaa7390e5a83ce6dd6ee50d3f0db49210e2e2 includes that.  Friday's release of plugin BOM will allow that version line to be removed.

File: pom.xml [hunk 4]
```diff
@@ -64,8 +65,11 @@
     <revision>5.5.3</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.baseline>2.440</jenkins.baseline>
-    <jenkins.version>${jenkins.baseline}.3</jenkins.version>
+    <!-- https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/ -->
+    <jenkins.baseline>2.462</jenkins.baseline>
+    <!-- TODO Replace with the standard jenkins.baseline references after LTS requires Java 17  -->
```
Comment (basil) [inline pom.xml:70]: This TODO can now be completed.

Comment (MarkEWaite) [reply pom.xml:70]: Completed in:

* #1685

**Pull request merged**